yarn-project/pxe/src/storage/note_data_provider/note_data_provider.ts

```diff
@@ -104,7 +133,30 @@ export class NoteDataProvider implements DataProvider {
     });
   }
 
-  public removeNotesAfter(blockNumber: number): Promise<void> {
+  /**
+   * Synchronizes notes and nullifiers to a specific block number.
+   *
+   * This method ensures that the state of notes and nullifiers is consistent with the
+   * specified block number. It restores any notes that were nullified after the given block
+   * and deletes any active notes created after that block.
+   *
+   * @param blockNumber - The block number to synchronize to (the reorg point)
```

Comment:
```suggestion
   * @param blockNumber - The new chain tip after reorg
```

I am being admittedly pedantic here but I didn't like the "synchronize to" because that makes me think we are going "up the chain" and not "down the chain".